### PR TITLE
Issue #249: Add commit hashes and messages to regression report

### PR DESCRIPTION
**Pull request by MEZk**
#249

### CODE REVIEW

**Comment by rnveach** [comment]
Can you show us a screenshot of new report index?

**Comment by MEZk** [comment]
@rnveach 

screenshot of a new report index

File: checkstyle-tester/diff.groovy [hunk 1]
```diff
@@ -269,6 +287,24 @@ def generateSummaryIndexHtml(diffDir) {
...
 }
 
+def printReportInfoSection(summaryIndexHtml, checkstyleBaseReportInfo, checkstylePatchReportInfo) {
+    summaryIndexHtml << ('<h6>')
+    if (checkstyleBaseReportInfo) {
+        summaryIndexHtml << "Base branch: $checkstyleBaseReportInfo.branch"
+        summaryIndexHtml << ('<br />')
+        summaryIndexHtml << "Base branch last commit SHA: $checkstyleBaseReportInfo.commitSha"
+        summaryIndexHtml << ('<br />')
+        summaryIndexHtml << "Base branch last commit msg: \"$checkstyleBaseReportInfo.commitMsg\""
```
Comment (rnveach) [inline checkstyle-tester/diff.groovy]: I think we should use full words in description, `msg` => `message`.

File: checkstyle-tester/diff.groovy [hunk 2]
```diff
@@ -269,6 +287,24 @@ def generateSummaryIndexHtml(diffDir) {
...
+        summaryIndexHtml << ('<br />')
+    }
+    summaryIndexHtml << "Patch branch: $checkstylePatchReportInfo.branch"
+    summaryIndexHtml << ('<br />')
+    summaryIndexHtml << "Patch branch last commit SHA: $checkstylePatchReportInfo.commitSha"
+    summaryIndexHtml << ('<br />')
+    summaryIndexHtml << "Patch branch last commit msg: \"$checkstylePatchReportInfo.commitMsg\""
```
Comment (rnveach) [inline checkstyle-tester/diff.groovy]: Same.

**Comment by rnveach** [comment]
Can we also add a empty line between `Patch branch: test` and previous line.

File: checkstyle-tester/diff.groovy (hunk 1 above)
Comment (MEZk) [inline checkstyle-tester/diff.groovy]: done

File: checkstyle-tester/diff.groovy (hunk 2 above)
Comment (MEZk) [inline checkstyle-tester/diff.groovy]: done

**Comment by MEZk** [comment]
@rnveach 
> Can we also add a empty line between Patch branch: test and previous line.

Sure. Done.

screenshot from 2017-07-10 22-50-08

**Pull request merged**